tools/serverpod_cli/lib/src/create/create.dart

```diff
@@ -94,363 +121,426 @@ Future<void> performCreate(
   var awsName = name.replaceAll('_', '-');
   var randomAwsId = math.Random.secure().nextInt(10000000).toString();
 
-  var projectDir = Directory(p.join(Directory.current.path, name));
-  if (projectDir.existsSync()) {
+  var serverpodDirs = ServerpodDirectories(
+    projectDir: Directory(p.join(Directory.current.path, name)),
+    name: name,
+  );
+  if (serverpodDirs.projectDir.existsSync()) {
     log.error('Project $name already exists.');
-    return;
+    return false;
   }
 
-  log.info(
-    'Creating project $name.',
-    style: const TextLogStyle(newParagraph: true),
-  );
+  if (template == ServerpodTemplateType.server) {
+    log.info(
+      '🚀 Creating Serverpod project "$name".',
+      type: TextLogType.init,
+    );
+  } else if (template == ServerpodTemplateType.module) {
+    log.info(
+      '📦 Creating Serverpod module "$name".',
+      type: TextLogType.init,
+    );
+  }
 
-  log.debug(
-    'Creating directory: ${projectDir.path}',
-    style: const TextLogStyle(
-      type: AbstractStyleType.bullet,
-    ),
-  );
-  projectDir.createSync();
+  bool success = await log.progress(
+      'Creating project directories.',
+      () => Future(() {
+            _createProjectDirectories(template, serverpodDirs);
+            return true;
+          }),
+      newParagraph: true);
 
-  var serverDir = Directory(p.join(projectDir.path, '${name}_server'));
-  log.debug(
-    'Creating directory: ${serverDir.path}',
-    style: const TextLogStyle(
-      type: AbstractStyleType.bullet,
-    ),
-  );
-  serverDir.createSync();
+  if (template == ServerpodTemplateType.server) {
+    success &= await log.progress(
+        'Writing project files.',
+        () => Future(() {
+              _copyServerTemplates(
+                serverpodDirs,
+                name: name,
+                awsName: awsName,
+                randomAwsId: randomAwsId,
+                dbPassword: dbPassword,
+                dbProductionPassword: dbProductionPassword,
+                dbStagingPassword: dbStagingPassword,
+              );
+              return true;
+            }));
 
-  var clientDir = Directory(p.join(projectDir.path, '${name}_client'));
-  log.debug(
-    'Creating directory: ${clientDir.path}',
-    style: const TextLogStyle(
-      type: AbstractStyleType.bullet,
-    ),
-  );
+    success &= await log.progress('Getting server package dependencies.',
+        () => CommandLineTools.dartPubGet(serverpodDirs.serverDir));
+    success &= await log.progress('Getting client package dependencies.',
+        () => CommandLineTools.dartPubGet(serverpodDirs.clientDir));
+    success &= await log.progress('Getting Flutter app package dependencies.',
+        () => CommandLineTools.flutterCreate(serverpodDirs.flutterDir));
```

Comment:
I think it would be a bit more readable to use `() {}` syntax on these too, even though they are one-liners.